pkg/serviceimport/map.go

```diff
@@ -33,18 +36,8 @@ type Map struct {
 	sync.RWMutex
 }
 
-func (m *Map) SelectIP(namespace, name string, checkCluster func(string) bool) (string, bool) {
-	queue, counter := func() ([]clusterInfo, *uint64) {
-		m.RLock()
-		defer m.RUnlock()
-
-		si, ok := m.svcMap[keyFunc(namespace, name)]
-		if !ok {
-			return nil, nil
-		}
-
-		return si.clustersQueue, &si.rrCount
-	}()
+func (m *Map) selectIP(si *serviceInfo, checkCluster func(string) bool) (string, bool) {
```

Comment:
```suggestion
func (m *Map) selectIP(queue []clusterInfo, counter *uint64, checkCluster func(string) bool) (string, bool) {
```

Also we don't need to return a bool anymore nor check for nil queue.